llvm/lib/Target/AMDGPU/SMInstructions.td

```diff
@@ -167,6 +167,20 @@ multiclass SM_Pseudo_Loads<RegisterClass baseClass,
   def _IMM : SM_Load_Pseudo <opName, baseClass, dstClass, IMM_Offset>;
   def _SGPR : SM_Load_Pseudo <opName, baseClass, dstClass, SGPR_Offset>;
   def _SGPR_IMM : SM_Load_Pseudo <opName, baseClass, dstClass, SGPR_IMM_Offset>;
+
+  // The constrained multi-dword load equivalents with early clobber flag at
+  // the dst operand. They are needed only for codegen and there is no need for
+  // their real opcodes.
+  let SubtargetPredicate = isGFX8Plus,
```

Comment:
Shouldn't really need a separate subtarget predicate for these from the base forms

Reply:
I guess, it is still ok to use `isGFX8Plus` predicate as this the only place we might need it (if introduce a new one).
Instead of adding a new subtarget predicate, what if I extend the comment to include xnack replay is supported for gfx8+ archs?

Reply:
The subtarget predicate isn't necessary at all. It's not wrong to over-constrain on targets without xnack